src/pu-emmc.c

```diff
@@ -881,5 +896,55 @@ pu_emmc_new(const gchar *device_path,
     if (!pu_emmc_parse_clean(self, root, error))
         return NULL;
 
+    /* Check if raw binary overwrites partitions */
+    if (self->partitions && self->raw) {
+        PuEmmcPartition *part = self->partitions->data;
+        gsize part_start = part->offset * self->device->sector_size;
+        gsize raw_start;
+        gsize raw_end;
+        struct stat st;
+        for (GList *b = self->raw; b != NULL; b = b->next) {
```

Comment:
Some vertical spacing here would be adequate to highlight logical pieces. There may be other places where you could add some, too.